source/uwp/SharedRenderer/lib/XamlHelpers.cpp

```diff
@@ -333,15 +333,28 @@ namespace AdaptiveCards::Rendering::Xaml_Rendering::XamlHelpers
 
             if (fallbackElementRenderer)
             {
-                fallbackControl = fallbackElementRenderer.Render(fallbackElement, renderContext, renderArgs);
-
-                renderedElement = fallbackElement;
+                try
+                {
+                    fallbackControl = fallbackElementRenderer.Render(fallbackElement, renderContext, renderArgs);
```

Comment:
Should we keep the fallbackControl nullcheck that triggers the call to RenderFallback here as well?

Reply:
I don't think so? at this point it will just be null(=dropped), so I think we're fine with this?

Reply:
I've checked how https://github.com/microsoft/AdaptiveCards/edit/main/samples/v1.2/Tests/ContainerDoubleFallback.json this works in nodejs renderer - and I don't think that works correct?
It will render `Hello World` but not as one of the items of the container rather as `substitution` of the container?
I thought the `notme` item should simply be dropped? since it doesn't have fallback?:)
@paulcam206 could you please clarify the desired falldown behavior?
Docs say:
![image](https://user-images.githubusercontent.com/21356912/221116900-4915ec34-0f9c-40d4-9f0c-273e1ad6affb.png)

so (if I understand this correctly) it says that if any of its children cannot be rendered -> then we execute the fallback -> meaning entire container just get substituted for a `TextBlock`? What if it was container within a container and that container had unknown type and wouldn't have a fallback? The outer container would become `TextBlock` again?

Reply:
Ah, fallback. This is by far one of the trickiest and least-used features of AC.

In the case of this particular test card, if the author wanted to replace the Container's `notme` element with some other element, `notme` would have `fallback` set on it. Since it doesn't, we look to the parent Container's `fallback`. We see that we do indeed have fallback (our `TextBlock`), and that gets subbed in for the `Container`.

Reply:
Also, just a heads up -- this feature was first developed in the shared cpp object model, so its implementation is closest to intended behavior (though there may be bugs). Last I remember, the NodeJS renderer isn't quite as good (particularly in the area of detecting ID collisions that occur as the result of fallback), so if their behavior differs the starting assumption should be that NodeJS doesn't quite have it right.

For reference, the id collision stuff is documented here:

https://github.com/microsoft/AdaptiveCards/blob/dd943981c76d71a8f4163b3e128b1abf17662475/source/shared/cpp/ObjectModel/ParseContext.cpp#L26-L144

Reply:
@paulcam206  thank you for the info!